islandora_compound_object.module

```diff
@@ -268,17 +268,94 @@ function islandora_compound_object_access(AbstractObject $object) {
 }
 
 /**
- * Replaces parent object thumbnail with child's thumbnail.
+ * Implements hook_compound_cmodel_islandora_derivative().
  */
-function islandora_compound_object_update_parent_thumbnail($parent) {
+function islandora_compound_object_compound_cmodel_islandora_derivative(AbstractObject $object = NULL, $ds_modified_params = array()) {
+  $module_path = drupal_get_path('module', 'islandora_solution_pack_compound');
+  return array(
+    array(
+      'source_dsid' => NULL,
+      'destination_dsid' => 'TN',
+      'function' => array('islandora_compound_object_update_parent_thumbnail'),
```

Comment:
Probably going to get into issues here, due to the [expected function signature](https://github.com/Islandora/islandora/blob/7.x/islandora.api.php#L686-L690), and [what we actually have](https://github.com/Islandora/islandora_solution_pack_compound/pull/126/files#diff-dacfa496dcdc86ad0174e734682f128bR342)... May be easiest to wrap things up in another function, to avoid changing the signature of the existing function?